### PR TITLE
[Feat/comment] 댓글 목록 조회 기능

### PR DESCRIPTION
**Pull request by adjoon1**
## #️⃣ Issue Number

- #13 

## 📝 요약(Summary)

댓글 목록 조회 기능 PR입니다.
기존에 IntelliJ에서만 테스트를 진행하고 push만 해놨던 상태인데 Postman 테스트까지 진행하였습니다.
처음 dev에서 팠던 브랜치라 다른 분들의 변경 사항이 적용되지 않은 상태입니다.
LocalDateTime을 Instant로 수정하기 전이라 이 부분은 댓글 삭제 기능에 반영되어 있습니다.

## 🛠️ PR 변경 사항
<-- 어떤 변경 사항이 있나요? -->

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경 등)
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 또는 설정 파일 수정
- [ ] 파일 또는 패키지 삭제

## 📸스크린샷 (선택)

스크린샷 2025-07-15 오전 2 10 53

스크린샷 2025-07-15 오전 2 11 07

스크린샷 2025-07-17 오전 5 20 29

## ✅ PR Checklist

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트)

## Summary by CodeRabbit

* **New Features**
  * Added a new API endpoint to retrieve a paginated list of comments for a specific article, supporting sorting and cursor-based pagination.
  * API responses include comment details such as content, author nickname, like count, and creation time.
* **Tests**
  * Introduced comprehensive tests for the new comment retrieval API and service logic, covering various scenarios and edge cases.

### CODE REVIEW

File: src/main/java/org/project/monewping/domain/comment/repository/CommentCustomRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    private final EntityManager em;
+    /**
+     * 기사 ID 및 커서 기반으로 댓글 목록을 조회합니다.
+     * 정렬 기준과 방향에 따라 커서 기반 페이지네이션을 수행하며,
+     * limit 개수만큼 댓글을 조회합니다.
+     */
+    @Override
+    public List<Comment> findComments(UUID articleId, String orderBy, String direction, String cursor, String after, int limit) {
+        StringBuilder sql = new StringBuilder("SELECT c FROM Comment c WHERE c.articleId = :articleId");
```
Comment (BackEunHo) [inline src/main/java/org/project/monewping/domain/comment/repository/CommentCustomRepositoryImpl.java:28]: Comment.java에서 논리 삭제된 댓글을 구분하기 위한 deleted 필드가 있는데, 댓글 ID로 댓글을 조회하는 해당 쿼리에서는 deleted 필드가 필터링되지 않은 것 같습니다. 논리 삭제 처리된 댓글도 함께 조회될 것 같습니다.

File: src/main/java/org/project/monewping/domain/comment/repository/CommentCustomRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        if (after != null) {
+            sql.append(" AND c.createdAt ");
+            sql.append("ASC".equalsIgnoreCase(direction) ? "> :after" : "< :after");
+        }
+
+        sql.append(" ORDER BY c.").append(orderBy).append(" ").append(direction.toUpperCase());
```
Comment (BackEunHo) [inline src/main/java/org/project/monewping/domain/comment/repository/CommentCustomRepositoryImpl.java:35]: orderBy와 direction에 문자열을 삽입해서 쿼리 구조를 변조할 수 있습니다. 그래서 orderBy와 direction을 허용된 값만 사용하도록 제한하면 좋을 것 같습니다. 서비스 또는 컨트롤러에서 설정할 수 있습니다.

예시:
List<String> allowedOrderBy = List.of("createdAt", "likeCount");
List<String> allowedDirection = List.of("ASC", "DESC");

if (!allowedOrderBy.contains(orderBy)) {
    throw new IllegalArgumentException("허용되지 않은 orderBy 값입니다.");
}
if (!allowedDirection.contains(direction.toUpperCase())) {
    throw new IllegalArgumentException("허용되지 않은 direction 값입니다.");
}
Injection 공격에 취약할 듯 합니다.

File: src/main/java/org/project/monewping/domain/comment/repository/CommentCustomRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        sql.append(", c.id ").append(direction.toUpperCase());
```
Comment (BackEunHo) [inline src/main/java/org/project/monewping/domain/comment/repository/CommentCustomRepositoryImpl.java:36]: 35번 줄의 내용과 같습니다.

**Review by BackEunHo — APPROVED** [review]

File: src/main/java/org/project/monewping/domain/comment/service/CommentServiceImpl.java
Comment (GeonMin02) [inline src/main/java/org/project/monewping/domain/comment/service/CommentServiceImpl.java:1]: getComments 메서드 내부 구현이 길어서 책임이 섞일 가능성이 높습니다
DB 조회나 커서 정보 계산 등
보조 메서드로 분리하면 테스트 및 유지보수에 용이합니다

또한, 로그 처리가 되어있다면 구조 동작 흐름 추적에 용이합니다

**Review by GeonMin02 — APPROVED** [review]
대부분의 리뷰는 코드래빗이 선수쳐서, 그 외에 존재하는 것중 할만한건 이렇게 인거같습니다
